cpp/src/arrow/parquet/parquet-io-test.cc

```diff
@@ -52,25 +53,104 @@ template <typename TestType>
 struct test_traits {};
 
 template <>
+struct test_traits<BooleanType> {
+  static constexpr ParquetType::type parquet_enum = ParquetType::BOOLEAN;
+  static constexpr LogicalType::type logical_enum = LogicalType::NONE;
+  static uint8_t const value;
+};
+
+const uint8_t test_traits<BooleanType>::value(1);
+
+template <>
+struct test_traits<UInt8Type> {
+  static constexpr ParquetType::type parquet_enum = ParquetType::INT32;
+  static constexpr LogicalType::type logical_enum = LogicalType::UINT_8;
```

Comment:
We should look at whether using the Parquet 2.0 logical types will cause problems in other readers. e.g. Impala doesn't have these ConvertedType enums

https://github.com/cloudera/Impala/blob/cdh5-trunk/common/thrift/parquet.thrift

@julienledem can you advise us here?

What other systems do is deal with converting Parquet's physical INT32 to smaller types through a layering of additional metadata -- for example, if a Hive or Impala table has `SMALLINT` or `TINYINT` type, then the int32 values are materilized into int16 or int8 slots